src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/TaskQuerySpecification.java

```diff
@@ -1,63 +1,167 @@
     }
 
-    public static Specification<TaskResource> searchByCaseId(List<String> caseIds) {
-        return (root, query, builder) -> builder.in(root.get(CASE_ID))
-            .value(caseIds);
+    public static Specification<TaskResource> searchByCaseIds(List<String> caseIds) {
+        if (!CollectionUtils.isEmpty(caseIds)) {
+            return (root, query, builder) -> builder.in(root.get(CASE_ID))
+                .value(caseIds);
+        }
+        return (root, query, builder) -> builder.conjunction();
     }
 
     public static Specification<TaskResource> searchByUser(List<String> users) {
-        return (root, query, builder) -> builder.in(root.get(ASSIGNEE))
-            .value(users);
+        if (!CollectionUtils.isEmpty(users)) {
+            return (root, query, builder) -> builder.in(root.get(ASSIGNEE))
+                .value(users);
+        }
+        return (root, query, builder) -> builder.conjunction();
     }
 
     public static Specification<TaskResource> searchByTaskId(String taskId) {
         return (root, query, builder) -> builder.equal(root.get(TASK_ID), taskId);
     }
 
     public static Specification<TaskResource> searchByTaskTypes(List<String> taskTypes) {
-        return (root, query, builder) -> builder.in(root.get(TASK_TYPE))
-            .value(taskTypes);
+        if (!CollectionUtils.isEmpty(taskTypes)) {
+            return (root, query, builder) -> builder.in(root.get(TASK_TYPE))
+                .value(taskTypes);
+        }
+        return (root, query, builder) -> builder.conjunction();
     }
 
     public static Specification<TaskResource> searchByWorkType(List<String> workTypes) {
-        return (root, query, builder) -> builder.in(root.get(WORK_TYPE).get("id"))
-            .value(workTypes);
+        if (!CollectionUtils.isEmpty(workTypes)) {
+            return (root, query, builder) -> builder.in(root.get(WORK_TYPE).get(WORK_TYPE_ID))
+                .value(workTypes);
+        }
+        return (root, query, builder) -> builder.conjunction();
+    }
+
+    public static Specification<TaskResource> extractState(SearchTaskRequest searchTaskRequest) {
+        final EnumMap<SearchParameterKey, SearchParameter> keyMap = asEnumMap(searchTaskRequest);
+        if (keyMap.get(SearchParameterKey.STATE) != null) {
+            final List<String> values = keyMap.get(SearchParameterKey.STATE).getValues();
+            final List<CFTTaskState> cftTaskStates = values.stream()
+                .filter(state -> state.trim().length() != 0)
+                .map(value -> CFTTaskState.valueOf(value.toUpperCase(Locale.ROOT))).collect(Collectors.toList());
+
+            return searchByState(cftTaskStates);
+        }
+
+        return (root, query, builder) -> builder.conjunction();
+    }
+
+    public static Specification<TaskResource> extractJurisdiction(SearchTaskRequest searchTaskRequest) {
+        final EnumMap<SearchParameterKey, SearchParameter> keyMap = asEnumMap(searchTaskRequest);
```

Comment:
a lot of these extractors duplicate this line since we pass in a searchTaskRequest

What this means:
Each time that we call one of these extractors we are looping through the entire searchtaskRequest object in order to map to the Enum map.

I think it would be cleaner to loop through it only once and build the map and then pass it onto these extractors.

Reply:
makes sense, will make changes.